core/tests/performance_framework/perf_domain.py

```diff
@@ -134,13 +134,101 @@ def _get_duration_millisecs(self, event_end, event_initial):
         return end_timestamp - initial_timestamp
 
     def get_page_load_time_millisecs(self):
-        """Returns the total page load time (in milliseconds)."""
+        """Returns the total page load time."""
         return self._get_duration_millisecs('loadEventEnd', 'fetchStart')
 
     def get_dom_ready_time_millisecs(self):
-        """Returns the total dom ready time (in milliseconds)."""
+        """Returns the time spent constructing the dom tree."""
         return self._get_duration_millisecs('domComplete', 'domInteractive')
 
     def get_request_time_millisecs(self):
-        """Returns the total request time (in milliseconds)."""
+        """Returns the time spent during request."""
         return self._get_duration_millisecs('responseEnd', 'requestStart')
+
+    def get_ready_start_time_millisecs(self):
+        """Returns the time consumed preparing the new page."""
+        return self._get_duration_millisecs('fetchStart', 'navigationStart')
+
+    def get_redirect_time_millisecs(self):
+        """Returns the time spent during redirection."""
+        return self._get_duration_millisecs('redirectEnd', 'redirectStart')
+
+    def get_appcache_time_millisecs(self):
+        """Returns the time spent for appcache."""
+        return self._get_duration_millisecs('domainLookupStart', 'fetchStart')
+
+    def get_unload_event_time_millisecs(self):
+        """Returns the time spent unloading documents."""
+        return self._get_duration_millisecs(
+            'unloadEventEnd', 'unloadEventStart')
+
+    def get_lookup_domain_time_millisecs(self):
+        """Returns the time spent for DNS query."""
```

Comment:
Can you be more specific? Is it for the main DNS query to resolve the page's domain, or is it all DNS queries?